tiledb/sm/c_api/api_validation.h

```diff
@@ -0,0 +1,328 @@
+/*
+ * @file   api_validation.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2021 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef API_VALIDATION_H
+#define API_VALIDATION_H
+
+#include "tiledb/sm/c_api/tiledb.h"
+#include "tiledb/sm/c_api/tiledb_struct_def.h"
+#include "tiledb/sm/enums/filter_type.h"
+
+/* ********************************* */
+/*         AUXILIARY FUNCTIONS       */
+/* ********************************* */
+
+/* Saves a status inside the context object. */
+static bool save_error(tiledb_ctx_t* ctx, const Status& st) {
+  // No error
+  if (st.ok())
+    return false;
+
+  // Store new error
+  ctx->ctx_->save_error(st);
+
+  // There is an error
+  return true;
+}
+
+static bool create_error(tiledb_error_t** error, const Status& st) {
+  if (st.ok())
+    return false;
+
+  (*error) = new (std::nothrow) tiledb_error_t;
+  if (*error == nullptr)
+    return true;
+  (*error)->errmsg_ = st.to_string();
+
+  return true;
+}
+
```

Comment:
If these are being moved into header for re-use, do we still want save_error() and create_error() to be 'static', or should they simply be prototyped with actual definition left in original .cc file?  
I speculatively assume that, for whatever reason, they were originally defined static to avoid them being available to other locations in code base.
Placed in a re-used header, they will be 'static'ally duplicated in all compiled modules including them, 
don't know if any/all linkers would attempt de-duping them or not...